econplayground/main/models.py

```diff
@@ -600,8 +598,11 @@ def __str__(self):
 
     def get_random(self):
         entry_list = list(self.questions.all())
-        pick = random.randrange(len(entry_list))  # nosec
-        return entry_list[pick]
+        if len(entry_list) > 0:
+            pick = random.randrange(len(entry_list))  # nosec
+            return entry_list[pick]
```

Comment:
Just a minor note, I think this code to get a random question can be improved upon a bit:
https://stackoverflow.com/a/62530224/173630

By using `self.questions.values_list('pk', flat=True)` instead of `self.questions.all()`, less data needs to be fetched from the database to get the random question.

Reply:
I've implemented the change in the student-view branch. This model is outdated.